app/services/ping.rb

```diff
@@ -47,6 +48,39 @@ def ping_database
       }
     end
 
+    def ping_cache
+      response = {}
+
+      if Rails.application.config.cache_store.first == :redis_cache_store
+        redis = Rails.cache.redis
+        response['servers'] = {}
+
+        if redis.respond_to?(:nodes)
+          nodes = redis.nodes
+        else
+          nodes = [redis]
+        end
+
+        nodes.each_with_index do |node, index|
+          start = Time.now
```

Comment:
It's better to use monotonic time, which is guaranteed not to jump back: `Process.clock_gettime(Process::CLOCK_MONOTONIC)`.

Reply:
Because it's in a loop?

Reply:
No, `Time.now` is not guaranteed to be consistently moving forward. If a time sync occurs (like with NTP) or with a leap second then it may jump back. I'm also not sure if DST can make it jump back/forward. Monotonic time is guaranteed to only tick forward. A quick search linked me to https://blog.codeminer42.com/the-monotonic-clock-and-why-you-should-care-about-it/ which is a bit more elaborate.

A good rule of thumb is that you never measure with `Time.now`.

Reply:
Let's do it first across the file -- https://github.com/theforeman/foreman/pull/9868